### PR TITLE
docs: update `docs/usage/routing/*`

### PR DESCRIPTION
**Pull request by JacobCoffee**
## Description

- Update `docs/routing/*`
Part of work in #3212

### CODE REVIEW

File: docs/usage/routing/handlers.rst [hunk 1]
```diff
@@ -2,320 +2,331 @@ Route handlers
...
+.. tip:: You can define a custom typing for your application state and then use it as a type instead of just using the
+    :class:`~.datastructures.state.State` class from Litestar
 
 Type annotations
 ----------------
 
-Litestar enforces strict type annotations. Functions decorated by a route handler **must** have all their kwargs and
-return value type annotated. If a type annotation is missing, an
-:class:`ImproperlyConfiguredException <litestar.exceptions.ImproperlyConfiguredException>` will be raised during the
+Litestar enforces strict type annotations. Functions decorated by a route handler **must** have all their
```
Comment (peterschutt) [inline docs/usage/routing/handlers.rst:139]: I wonder if there's a link we can use for the term "annotations" like we've done for other things like decorators etc. Maybe https://peps.python.org/pep-0484/#type-definition-syntax?

File: docs/usage/routing/handlers.rst [hunk 2]
```diff
@@ -2,320 +2,331 @@ Route handlers
...
+:term:`kwargs <argument>` and return value type annotated.
```
Comment (peterschutt) [inline docs/usage/routing/handlers.rst:140]:
```suggestion
:term:`arguments <argument>` and return value type annotated.
```
I think this is more correct.

File: docs/usage/routing/handlers.rst [hunk 3]
```diff
@@ -2,320 +2,331 @@ Route handlers
...
 
-
-#. to ensure best practices
-#. to ensure consistent OpenAPI schema generation
-#. to allow Litestar to compute during the application bootstrap all the kwargs required by a function
-
+#. To ensure best practices
+#. To ensure consistent OpenAPI schema generation
+#. To allow Litestar to compute during the application bootstrap all the :term:`kwargs <argument>` required by a
+   function
```
Comment (peterschutt) [inline docs/usage/routing/handlers.rst:149]:
```suggestion
#. To allow Litestar to compute the :term:`arguments <argument>` required by a function during application bootstrap
```

File: docs/usage/routing/handlers.rst [hunk 4]
```diff
@@ -2,320 +2,331 @@ Route handlers
...
 
 
-   @route(path="/some-path", http_method=[HttpMethod.GET, HttpMethod.POST])
-   async def my_endpoint() -> None: ...
+    @route(path="/some-path", http_method=[HttpMethod.GET, HttpMethod.POST])
+    async def my_endpoint() -> None: ...
 
-As mentioned above, ``route`` does is merely an alias for ``HTTPRouteHandler``\ , thus the below code is equivalent to the one
-above:
+As mentioned above, :func:`@route() <.handlers.route>` does is merely an alias for ``HTTPRouteHandler``,
```
Comment (peterschutt) [inline docs/usage/routing/handlers.rst:168]:
```suggestion
As mentioned above, :func:`@route() <.handlers.route>` is merely an alias for ``HTTPRouteHandler``,
```

**Review by peterschutt — COMMENTED** [review]
WIP

File: docs/usage/routing/handlers.rst [hunk 5]
```diff
@@ -2,320 +2,331 @@ Route handlers
...
-:class:`BaseRouteHandler <.handlers.base.BaseRouteHandler>`. These classes are optimized
-descriptor classes that record all the data necessary for the given function or method - this includes a modelling of
-the function signature, which allows for injection of kwargs and dependencies, as well as data pertinent to OpenAPI
-spec generation.
+In the above example, the :term:`decorator` includes all the information required to define the endpoint operation for
+the combination of the path ``"/"`` and the HTTP verb ``GET``. In this case it will be a HTTP response with a
+``Content-Type`` header of ``text/plain``.
 
+What the :term:`decorator` does is wrap the function or method within a class instance that inherits from
+:class:`~.handlers.base.BaseRouteHandler`. These classes are optimized descriptor classes that record all the data
```
[the remark this answers is not in the document]
Comment (JacobCoffee) [reply docs/usage/routing/handlers.rst:24]: If it helps as an end user reading this, i was lost.

File: docs/usage/routing/handlers.rst (hunk 1 above)
Comment (JacobCoffee) [reply docs/usage/routing/handlers.rst:139]: Im not sure here, i believe i tried but the base PR this comes from is a little old so i may be misremembering.

Comment (JacobCoffee) [reply docs/usage/routing/handlers.rst:139]: oh not i didnt try at all or hard enough (or both)
Is this what you mean?
https://docs.python.org/3/glossary.html#term-annotation

File: docs/usage/routing/parameters.rst [hunk 6]
```diff
@@ -37,187 +36,176 @@ Currently, the following types are supported:
...
-    :language: python
-
-
-In the above example, :func:`Parameter <.params.Parameter>` is used to restrict the value of ``version`` to a range
-between 1 and 10, and then set the ``title``, ``description``, ``examples``, and ``externalDocs`` sections of the
-OpenAPI schema.
+    :caption: Adding extra validation and documentation to a path parameter
 
+In the above example, :func:`~.params.Parameter` is used to restrict the value of :paramref:`~.params.Parameter.version`
+to a range between 1 and 10, and then set the :paramref:`~.params.Parameter.title``,
```
Comment (JacobCoffee) [inline docs/usage/routing/parameters.rst:75]:
```suggestion
to a range between 1 and 10, and then set the :paramref:`~.params.Parameter.title`,
```

File: docs/usage/routing/parameters.rst
Comment (JacobCoffee) [inline docs/usage/routing/parameters.rst:1]: overlinking of the term `parameter` here..

File: docs/usage/routing/handlers.rst (hunk 1 above)
Comment (peterschutt) [reply docs/usage/routing/handlers.rst:139]: Yeh that'll do!

File: docs/usage/routing/overview.rst
[the remark this answers is not in the document]
Comment (JacobCoffee) [reply docs/usage/routing/overview.rst:1]: any ideas @litestar-org/maintainers

**Comment by JacobCoffee** [comment]
Besides the confusing the raised by @peterschutt and the ask of renaming the internal section since the top level is also `Routing` this is ready

File: docs/usage/routing/handlers.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (peterschutt) [reply docs/usage/routing/handlers.rst:24]: I think we just remove that terminology - its not doing beginners any favors by being there.

File: docs/usage/routing/overview.rst
[the remark this answers is not in the document]
Comment (peterschutt) [reply docs/usage/routing/overview.rst:1]: B/c we  have "Routing" on the index page also?

[the remark this answers is not in the document]
Comment (peterschutt) [reply docs/usage/routing/overview.rst:1]: what about "Overview"?

**Review by peterschutt — APPROVED** [review]

**Pull request merged**